// @browsercash/sdk/src/resources/browser/browser.ts
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.

import { APIResource } from '../../core/resource';
import * as SessionAPI from './session';
import {
  Session,
  SessionCreateParams,
  SessionCreateResponse,
  SessionGetParams,
  SessionGetResponse,
  SessionListParams,
  SessionListResponse,
  SessionStopParams,
  SessionStopResponse,
} from './session';

export class Browser extends APIResource {
  session: SessionAPI.Session = new SessionAPI.Session(this._client);
}

Browser.Session = Session;

export declare namespace Browser {
  export {
    Session as Session,
    type SessionCreateResponse as SessionCreateResponse,
    type SessionListResponse as SessionListResponse,
    type SessionGetResponse as SessionGetResponse,
    type SessionStopResponse as SessionStopResponse,
    type SessionCreateParams as SessionCreateParams,
    type SessionListParams as SessionListParams,
    type SessionGetParams as SessionGetParams,
    type SessionStopParams as SessionStopParams,
  };
}
